atmo/clusters/provisioners.py

```diff
@@ -7,10 +7,25 @@
 class ClusterProvisioner(Provisioner):
     log_dir = 'clusters'
 
+    def __init__(self):
+        super(ClusterProvisioner, self).__init__()
+        # the S3 URI to the script-runner jar
+        self.jar_uri = (
+            's3://%s.elasticmapreduce/libs/script-runner/script-runner.jar' %
+            self.config['AWS_REGION']
+        )
+
+        # the S3 URI to the zeppelin setup step
+        self.zeppelin_uri = (
+            's3://%s/steps/zeppelin/zeppelin.sh' %
+            self.config['SPARK_EMR_BUCKET']
+        )
+
     def job_flow_params(self, *args, **kwargs):
         params = super(ClusterProvisioner, self).job_flow_params(*args, **kwargs)
         # don't auto-terminate the cluster
         params.setdefault('Instances', {})['KeepJobFlowAliveWhenNoSteps'] = True
+        params['Applications'].append({'Name': 'Zeppelin'})
```

Comment:
Let's use `params.setdefault('Applications', []).append({'Name': 'Zeppelin'})` here to code a bit defensively in case the parent `job_flow_params` stops returning a `Applications` item. 🐍